app/main.py

```diff
@@ -1,2 +1,148 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    _DEFAULT_INITIAL_CAPACITY = 8
+    _DEFAULT_LOAD_FACTOR = 2 / 3
+    _DEFAULT_CAPACITY_MULTIPLIER = 2
+
+    @dataclass
+    class Node:
+        key: Hashable
+        value: Any
+
+    def __init__(
+            self,
+            initial_capacity: int = _DEFAULT_INITIAL_CAPACITY,
```

Comment:
Actually, I think it makes sense since the name of the task is "custom dict" (what if I need a dictionary with a load factor 1/2 instead of the 2/3, and next time I'll need it with capacity multiplier 3 to satisfy my custom requirements?).

Reply:
This task is more about to learn about dict internal structure, there are no strict guidelines about what should be in `__init__` method. What I'm trying to say is that parametrizing such an important data structure will make it unstable and unpredictable.